src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java

```diff
@@ -261,6 +264,16 @@ private boolean isMissingRequiredBillingProvider(
           "PAYG eligible subscription with subscriptionId:{} and subscription_number:{} has no billing provider.",
           subscription.getSubscriptionId(),
           subscription.getSubscriptionNumber());
+    }
+  }
+
+  private boolean isNewContractWithoutExistingSubscription(
+      org.candlepin.subscriptions.db.model.Subscription subscription,
+      Optional<org.candlepin.subscriptions.db.model.Subscription> existingSubscription) {
+    if (subscription.getOffering().isMetered() && existingSubscription.isEmpty()) {
```

Comment:
This will skip all metered subscriptions. We should instead only check contractenabled subscriptions and skip them.

Reply:
For the sake of clarity, Rosa is metered and contract-enabled; therefore, we can disregard the subscriptions from the UMB message. However, RHEL ELS Payg is metered but not contract-enabled, and we would have skipped all of the subscriptions in that scenario.